packages/astro/src/content/loaders/glob.ts

```diff
@@ -190,14 +190,51 @@ export function glob(globOptions: GlobOptions): Loader {
 			}
 
 			const limit = pLimit(10);
+			const skippedFiles: Array<string> = [];
+
+			const contentDir = new URL('content/', settings.config.srcDir);
+
+			function isInContentDir(file: string) {
+				const fileUrl = new URL(file, baseDir);
+				return fileUrl.href.startsWith(contentDir.href);
+			}
+
+			function isConfigFile(file: string) {
+				const configFiles = ['config.js', 'config.ts', 'config.mjs'];
+				const fileUrl = new URL(file, baseDir);
+				return configFiles.some(
+					(configFile) => new URL(configFile, contentDir).href === fileUrl.href
+				);
+			}
+
 			await Promise.all(
 				files.map((entry) =>
 					limit(async () => {
+						if (isConfigFile(entry)) {
+							return;
+						}
+						if (isInContentDir(entry)) {
+							skippedFiles.push(entry);
+							return;
+						}
```

Comment:
Can we move this one block upwards, e.g.:

```ts
files.map((entry) => {
  if (isConfigFile(entry)) {
    return
  }
  if (isInContentDir(entry)) {
    skippedFiles.push(entry);
    return;
  }
  return limit(async () => { ... })
})
```

That way there's less pressure for `p-limit` to track promises and lesser promises gets created too.